lib/tasks/sample_data.rake

```diff
@@ -16,111 +18,129 @@ namespace :sample_data do
         name: "Band #{n}",
         phone: "9999-9999"
       )
+
+      update_or_create_user(
+        email: "user-band-#{n}@email.com",
+        password: "password",
+        band_id: band.id
+      )
     end
   end
 
   desc "Creates musicians"
   task musicians: :environment do
 
     5.times do |n|
-      update_or_create_musician(
-        email: "musician0#{n}@email.com",
-        password: "password",
-        password_confirmation: "password",
+      musician = update_or_create_musician(
         name: "Musician 0#{n}",
         last_name: "McMusic",
         phone: "9999-9999",
         identification: "0-0000-0000",
         band_id: Band.first.id
       )
+
+      user = update_or_create_user(
+        email: "user-musician-b1-#{n}@email.com",
+        password: "password",
+        musician_id: musician.id
+      )
     end
 
     5.times do |n|
-      update_or_create_musician(
-        email: "musician1#{n}@email.com",
-        password: "password",
-        password_confirmation: "password",
+      musician = update_or_create_musician(
         name: "Musician 1#{n}",
         last_name: "McMusic",
         phone: "9999-9999",
         identification: "0-0000-0000",
         band_id: Band.second.id
       )
+
+      user = update_or_create_user(
+        email: "user-musician-b2-#{n}@email.com",
+        password: "password",
+        musician_id: musician.id
+      )
     end
 
     5.times do |n|
-      update_or_create_musician(
-        email: "musician2#{n}@email.com",
-        password: "password",
-        password_confirmation: "password",
+      musician = update_or_create_musician(
         name: "Musician 2#{n}",
         last_name: "McMusic",
         phone: "9999-9999",
         identification: "0-0000-0000"
       )
+
+      user = update_or_create_user(
+        email: "user-musician-#{n}@email.com",
+        password: "password",
+        musician_id: musician.id
+      )
     end
   end
 
   desc "Creates events"
   task events: :environment do
-    10.times do |n|
-      update_or_create_event(
-        name: "Event 1#{n}",
-        date: Date.tomorrow,
-        description: "description 1#{n}",
-        client_id: Client.first.id
-      )
-    end
-
-    10.times do |n|
+    20.times do |n|
       update_or_create_event(
-        name: "Event 2#{n}",
+        name: "Event #{n}",
         date: Date.tomorrow,
-        description: "description 2#{n}",
-        client_id: Client.second.id
+        time: Time.now,
+        email: "event#{n}@email.com",
+        phone: "9999-9999",
+        description: "description #{n}",
+        client_id: 1 + rand(Client.all.count),
+        event_type_id: 1 + rand(EventType.all.count)
       )
     end
   end
 
   def update_or_create_client(attributes)
-    client = Client.find_or_initialize_by(email: attributes.delete(:email))
+    client = Client.find_or_initialize_by(name: attributes.delete(:name))
     client.update_attributes(attributes)
+    return client
```

Comment:
As you say my man, I indeed know that Ruby does return the last line and as always, there's a reasonable explanation for my actions :smile: 

The method `update_attributes` you see in line 121 does not return the object it's updating, it returns a `true` value if it succeeds or a `false` value if it fails, that's why it is commonly used like this:

```
if @foo.update_attributes(params)
    # life is good
else
    # something is wrong
end
```
Sources: 
- http://apidock.com/rails/ActiveRecord/Base/update_attributes

- http://apidock.com/rails/ActiveRecord/Persistence/update_attributes

However, if you check lines `#25 #39 #56 #64 #72 #80 #88 #95`, what I need from the methods `update_or_create_band`, `update_or_create_musician` and `update_or_create_client` is not a `true` or `false` value, but the object that was just created/updated, and that my friend, is why I need that `return` statement exactly where it is :grin: